### PR TITLE
(SERVER-2352) Remove mention of old features in the docs

### PR DESCRIPTION
**Pull request by clairecadman**
This commit removes or deprecates rack master, passenger and Ruby master in the docs.

### CODE REVIEW

File: documentation/puppetserver_vs_passenger.markdown [hunk 1]
```diff
@@ -4,6 +4,10 @@ title: "Puppet Server: Puppet Server vs. Apache/Passenger Puppet Master"
 canonical: "/puppetserver/latest/puppetserver_vs_passenger.html"
 ---
 
+> ## Important: Deprecation warning
+>
+> The apache/passenger Puppet master is deprecated and will be removed in a future Puppet release.
+
```
Comment (Magisus) [inline documentation/puppetserver_vs_passenger.markdown]: Just to make sure, it's not already gone as of Puppet 6? Just want to make sure docs around it aren't only relevant for upgrades.

Comment (gguillotte) [reply documentation/puppetserver_vs_passenger.markdown]: I think Webrick support and the `puppet master` command were removed in PUP-8544 and PUP-8591.

It doesn't look like the docs for it were removed from Puppet 6: https://puppet.com/docs/puppet/6.0/services_master_webrick.html

PUP-9055 is a complication.

Comment (justinstoller) [reply documentation/puppetserver_vs_passenger.markdown]: Yeah, apache/passenger masters were deprecated in Puppet 4 and finally removed in 6. This file is no longer applicable.

**Comment by justinstoller** [comment]
Can you target this at 6.0.x?

**Comment by justinstoller** [comment]
Hey @clairecadman 	this needs a rebase to work with 6.0.x. Are you comfortable doing that in git, or do you want me to do it?

**Pull request merged**